src/driver-hidpp20.c

```diff
@@ -1509,7 +1509,6 @@ hidpp20drv_remove(struct ratbag_device *device)
 	if (drv_data->profiles)
 		hidpp20_onboard_profiles_destroy(drv_data->profiles);
 	free(drv_data->led_infos.color_leds_8070);
-	free(drv_data->led_infos.color_leds_8071);
 	free(drv_data->controls);
```

Comment:
whoah, yeah, that's a double free, that info is in a union. Though really we need to wrap the `free()` calls into a capability check, similar to `hidpp20drv_read_color_leds()`.

Reply:
We can make a macro to free and make the value NULL.

Reply:
@FFY00  I will remove the second commit if you decide to fix it later.

Reply:
> We can make a macro to free and make the value NULL.

we'd still rely on this struct to be free-able with a single call. better to be correct here and add the `if` so we don't introduce new bugs if we ever extend this.